### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by Apatakiti**
### Check the usage of Decorator pattern in Ruby
- Think about how you can use two decorators in order to capitalize and trim people's names.
- In the following steps you will get instructions for implementing the Decorator design pattern for this case.
### Create a class Nameable.
- Implement a method called correct_name that will raise a NotImplementedError.
- Turn your Person class to Nameable
- Make sure that your Person class inherits from Nameable
- Make sure that this class has a method correct_name implemented. It should simply return the name attribute.
- Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
### Create a class that inherits from the base Decorator class.
- Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- For the TrimmerDecorator:
### Create a class that inherits from the base Decorator class.
- Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

- puts capitalized_trimmed_person.correct_name

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
+require './decorator'
+
 class Person
   attr_accessor :name, :age, :id
```
Comment (Qoosim) [inline person.rb]: - [x] `Person` class is supposed to inherit from Nameable class and should also have a method `correct_name` that returns name. It should look something similar to this.

```
    class Person < Nameable
       .
       .

      def correct_name
         name
      end
  .
  .

```

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle:

Hi @Ademola-coding , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  Implemented some project requirements. :heavy_check_mark: 
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct Gitflow. :heavy_check_mark: 




## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by divinecharlotte — APPROVED** [review]
# Hello @Ademola-coding      😃 🥳 🎉 🎈 

Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 
### To Highlight 💯

- Good implementation of the  requested changes✔️
- Descriptive pr✔️






## Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (Apatakiti) [reply person.rb]: Done thanks

**Pull request merged**